cdap-ui/app/cdap/components/CaskWizards/PublishPipeline/index.js

```diff
@@ -105,7 +114,24 @@ export default class PublishPipelineWizard extends Component {
     };
     let currentNamespace = NamespaceStore.getState().selectedNamespace;
     let draftId;
-    if (this.props.input.action.type === 'create_pipeline_draft') {
+    if (this.props.isUsecase) {
+      return MyPipelineApi
+        .publish({
+          namespace: currentNamespace,
+          appId: name
+          }, {
+            artifact,
+            config: pipelineConfig
+          }
+        )
+        .map((res) => {
+          if (this.props.input && this.props.input.isLastStepInMarket) {
```

Comment:
I'm not sure about this. We have a market specific wizard for publish pipeline and yet we have `isLastStepInMarket` in `PublishPipeline` wizard?